chain/chunks/src/lib.rs

```diff
@@ -1273,7 +1276,7 @@ impl ShardsManager {
             };
         }
 
-        if header.shard_id() >= self.epoch_manager.num_shards(&epoch_id)? {
+        if !self.epoch_manager.shard_ids(&epoch_id)?.contains(&header.shard_id()) {
```

Comment:
This is a bit slower than before and it's hinting that the shards should actually be stored as a set rather then a vector. Totally not for this PR, just general thoughts for the future.

Reply:
Yes I noticed the same thing.  In general, I wonder how often we can expect number of shards of change and if the information could be cached.  Still, we should do some performance analysis before introducing such complexities.

Reply:
In theory it shard layout could change every epoch or every two epochs - quite a long time to begin with. In practice it's more likely to be much less frequent. The reason why it cannot change more often is becuase it's tied to the protocol version which can only change at epoch boundary.